src/components/Paginator.tsx

```diff
@@ -0,0 +1,133 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import range from 'lodash.range';
+import Icon from './Icon';
+import Page from './Paginator/Page';
+import Pagination from './Pagination';
+import ShortcutLink from './Paginator/ShortcutLink';
+import State from './Paginator/State';
+import Summary from './Paginator/Summary';
+
+const DEFAULT_PER_PAGE = 20;
+
+type LinkProps = {
+  disabled?: boolean,
+  page: number,
+  onClick: (page: number) => void,
+}
+
+const FirstPageLink: React.FunctionComponent<LinkProps> = ({ disabled, page, onClick }) => (
+  <ShortcutLink name="first" page={page} disabled={disabled} onClick={onClick}>
+    <Icon name="angle-double-left" />
+  </ShortcutLink>
+);
+
+const PrevPageLink: React.FunctionComponent<LinkProps> = ({ disabled, page, onClick }) => (
+  <ShortcutLink name="previous" page={page} disabled={disabled} onClick={onClick}>
+    <Icon name="angle-left" />
+  </ShortcutLink>
+);
+
+const NextPageLink: React.FunctionComponent<LinkProps> = ({ disabled, page, onClick }) => (
+  <ShortcutLink name="next" page={page} disabled={disabled} onClick={onClick}>
+    <Icon name="angle-right" />
+  </ShortcutLink>
+);
+
+const LastPageLink: React.FunctionComponent<LinkProps> = ({ disabled, page, onClick }) => (
+  <ShortcutLink name="last" page={page} disabled={disabled} onClick={onClick}>
+    <Icon name="angle-double-right" />
+  </ShortcutLink>
+);
+
+const linkPropTypes = {
+  disabled: PropTypes.bool,
+  page: PropTypes.number.isRequired,
+  onClick: PropTypes.func.isRequired
+};
+
+FirstPageLink.propTypes = linkPropTypes;
+PrevPageLink.propTypes = linkPropTypes;
+NextPageLink.propTypes = linkPropTypes;
+LastPageLink.propTypes = linkPropTypes;
+
+type PaginatorProps = {
+  currentPage: number,
+  onClick: (page: number) => void,
+  perPage?: number,
+  size?: 'sm' | 'lg',
+  summary: React.ReactNode,
```

Comment:
The `Paginator` component has built-in handling for the absence of the `summary` prop:
https://github.com/appfolio/react-gears/blob/1adbf65196bce045bc1596a602db6a4e3998a849/src/components/Paginator.tsx#L119
and there are numerous tests in `Paginator.spec.js` that do not pass the `summary` prop:
https://github.com/mycase/react-gears/blob/1adbf65196bce045bc1596a602db6a4e3998a849/test/components/Paginator.spec.js#L8-L189
so it seems that the `PaginatorProps` type should define `summary` as an optional prop.

Reply:
This is fixed in https://github.com/appfolio/react-gears/pull/814